lib/marqeta/api_caller.rb

```diff
@@ -1,61 +1,66 @@
 require 'json'
-require 'rest-client'
+require 'faraday'
 require 'marqeta/errors'
 
 module Marqeta
   class ApiCaller
     def initialize(endpoint, params = {})
       @endpoint = endpoint
-      @endpoint += "?#{URI.encode_www_form(params)}" if params.any?
+      @connection = Faraday.new(url: "http://" + Marqeta.configuration.base_url) do |conn|
+        conn.request :authorization, :basic, Marqeta.configuration.username, Marqeta.configuration.password
+        conn.response :logger, logger
```

Comment:
there is a [logger middleware](https://lostisland.github.io/faraday/#/middleware/included/logging) for faraday which will pass all of the request/response details directly to the logger. as a result, we don't need our manual logging calls, and don't need to test that they're happening.